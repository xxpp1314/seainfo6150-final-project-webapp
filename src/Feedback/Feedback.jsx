import React, { useState } from "react";
import styles from "./Feedback.module.css";

const Feedback = () => {
  const [submittedForm, setSubmittedForm] = useState();
  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedForm(data);
  };

  return (
    <div className={styles.grid}>
      <div>
      {
        submittedForm ? (
          <div className={styles.thankyou}>          
              <h1>Thank you for your valuable feedback</h1>
          </div>
        ) : (
         <div className={styles.feedback}>
            <h2 className={styles.h2}>Your opinions and suggestions are highly valuable. Looking forward to your feedback.</h2><br />
            <div>
            <form onSubmit={onSubmit} className={styles.text}><br />
                How would you rate this website?<br /><br />
                <input type="radio" id="1" name="place" value="1" />
                <label htmlFor="1">1 </label>
                <input type="radio" id="2" name="place" value="2" />
                <label htmlFor="2">  2 </label>
                <input type="radio" id="3" name="place" value="3" />
                <label htmlFor="3">  3 </label>
                <input type="radio" id="4" name="place" value="4" />
                <label htmlFor="4">  4 </label>
                <input type="radio" id="5" name="place" value="5" />
                <label htmlFor="5">  5 </label><br /><br /><br />

                <label htmlFor="myTextId">Do you think which information help you find a suitable pet?</label><br />
                <input  type="text" name="text1" id="text1" className={styles.textField} required/><br /><br />

                <label htmlFor="myTextId">Which part do you think we need to improve?</label><br />
                <input type="text" name="text2" id="text2" className={styles.textField} /><br /><br />

                <label htmlFor="myTextId">Do you want to share your experience when you use our website?</label><br />
                <input type="text" name="text3" id="text3" className={styles.textField} /><br /><br />

                <label htmlFor="myTextId">Is there anything else do you want to share with us?</label><br />
                <input type="text" name="text3" id="text3" className={styles.textField} /><br /><br />

                <input type="submit" value="Submit" className={styles.button}/><br /><br />
                </form>

            </div>
        </div>
        )
      }
      </div>
    </div>
  )
}

export default Feedback;