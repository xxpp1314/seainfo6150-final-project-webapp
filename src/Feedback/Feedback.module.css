.feedback{
    background-image: linear-gradient(to right,rgb(83, 15, 15), rgb(49, 12, 25) 70%);
    padding: 5%;
    box-shadow: 0px 0px 10px 0px rgba(204, 0, 0, 0.5);    
    border: 0px solid #666;
    display: flex;
    flex-direction: column;
    align-items: center;
    font-family: 'Arima Madurai', cursive;
    font-size: medium;
}

.thankyou{
    padding: 15%;
    text-align: center;
    color:  rgb(201, 187, 187);
    box-shadow: 0px 0px 10px 0px rgba(204, 0, 0, 0.5);    
    border: 0px solid #666;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.grid{
    display: grid;
    grid-template-columns: 1fr;
    align-items: center;
    background-image: linear-gradient(to right,rgb(83, 15, 15), rgb(49, 12, 25) 70%);
    padding: 5%;
    box-shadow: 0px 0px 10px 0px rgba(204, 0, 0, 0.5);
}

.textField {
    height: 50px;
    width: 300px;
    background-color: rgb(243, 234, 234);
}

@media screen and (min-width: 100px) {
    .textField{
      height:100%;
      width:100%;
    }
}

@media screen and (min-width: 800px) {
    .textField{
        height: 60px;
        width: 400px;
    }
}

.button{
    align-self: center;
    font-weight: bolder;
    background-color: rgb(97, 50, 50);
    border-color: rgb(243, 222, 222);
    height: 30px;
    width: 200px;
    color:rgb(243, 222, 222);
  }

.h2{
    text-align: center;
    color:rgb(243, 222, 222);
}  

.text{
    color:rgb(243, 222, 222);
}