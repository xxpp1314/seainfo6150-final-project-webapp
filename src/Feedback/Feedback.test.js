import React from "react";
import { render } from "@testing-library/react";
import Feedback from "./Feedback.jsx";

it("renders the Feedback Page correctly", () => {
  const { container } = render(
    <Feedback />
  );
  expect(container).toMatchSnapshot();
});
