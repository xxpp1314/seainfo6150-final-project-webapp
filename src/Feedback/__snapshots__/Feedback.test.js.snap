// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Feedback Page correctly 1`] = `
<div>
  <div
    class="grid"
  >
    <div>
      <div
        class="feedback"
      >
        <h2
          class="h2"
        >
          Your opinions and suggestions are highly valuable. Looking forward to your feedback.
        </h2>
        <br />
        <div>
          <form
            class="text"
          >
            <br />
            How would you rate this website?
            <br />
            <br />
            <input
              id="1"
              name="place"
              type="radio"
              value="1"
            />
            <label
              for="1"
            >
              1 
            </label>
            <input
              id="2"
              name="place"
              type="radio"
              value="2"
            />
            <label
              for="2"
            >
                2 
            </label>
            <input
              id="3"
              name="place"
              type="radio"
              value="3"
            />
            <label
              for="3"
            >
                3 
            </label>
            <input
              id="4"
              name="place"
              type="radio"
              value="4"
            />
            <label
              for="4"
            >
                4 
            </label>
            <input
              id="5"
              name="place"
              type="radio"
              value="5"
            />
            <label
              for="5"
            >
                5 
            </label>
            <br />
            <br />
            <br />
            <label
              for="myTextId"
            >
              Do you think which information help you find a suitable pet?
            </label>
            <br />
            <input
              class="textField"
              id="text1"
              name="text1"
              required=""
              type="text"
            />
            <br />
            <br />
            <label
              for="myTextId"
            >
              Which part do you think we need to improve?
            </label>
            <br />
            <input
              class="textField"
              id="text2"
              name="text2"
              type="text"
            />
            <br />
            <br />
            <label
              for="myTextId"
            >
              Do you want to share your experience when you use our website?
            </label>
            <br />
            <input
              class="textField"
              id="text3"
              name="text3"
              type="text"
            />
            <br />
            <br />
            <label
              for="myTextId"
            >
              Is there anything else do you want to share with us?
            </label>
            <br />
            <input
              class="textField"
              id="text3"
              name="text3"
              type="text"
            />
            <br />
            <br />
            <input
              class="button"
              type="submit"
              value="Submit"
            />
            <br />
            <br />
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
